### ISSUE TITLE
Upgrade to Django 2

### ISSUE THREAD

**Issue by Bomme** [MTG/freesound#1660]
After we reach compatibility with Python 3 (#1083), we can upgrade to the next major version of Django

The [Django upgrade guide](https://docs.djangoproject.com/en/2.0/howto/upgrade-version/) states:
> If you’re upgrading through more than one feature version (e.g. A.B to A.B+2), it’s usually easier to upgrade through each feature release incrementally (A.B to A.B+1 to A.B+2) rather than to make all the changes for each feature release at once. For each feature release, use the latest patch release (A.B.C).
>
>The same incremental upgrade approach is recommended when upgrading from one LTS to the next.

Before we attempt the upgrade we should make sure to:

- [x] Read the [release notes](https://docs.djangoproject.com/en/2.2/releases/) for each ‘final’ release from the one after your current Django version, up to and including the version to which you plan to upgrade.
- [x]  upgrade to the latest version Django-related dependencies 
- [x] resolve any deprecation warnings raised by the current version of Django
- [x] follow deployment steps suggested by Django: https://docs.djangoproject.com/en/2.2/howto/upgrade-version/

**Comment by ffont** [comment]
Hi, upgrading Django is going to be awesome!
The most difficult part will most likely be the nightmare of compatibility between FS dependencies and versions of django. However, to help in this process I went through all FS django-related dependencies and checked with which versions of Django they are compatible. Below I copy a list of my conclusions. I indicate versions of the dependency that are compatible with django 1.11, django 2.2 and 3.2+. I guess first step would be to upgrade all dependencies to the latest 1.11 compatible version (which in most cases is "guaranteed" that it is also 2.2 compatible and in some other cases I'm sure it will also be). Then upgrade django to 2.2. Then upgrade dependencies to latests 2.2 compatible version (which will hopefully be also compatible with django 3.2+). Then upgrade django to 3.2, etc...

Anyway, here is the data I collected:

> django-amazon-ses==3.0.1  

Django compatibility not well specified, our current version will hopefully support Django 2.2
4.0.1 (latest) compatible with Django 2.2+

> django-cors-headers==2.5.3

3.2.1 is the last compatible with Django 1.11 (and is also compatible with 2.2)
3.11.0 is the last compatible with Django 2.2
3.13.0 (latest) compatible with Django 3.2+

> django-debug-toolbar==1.7

2.2.1 is the last compatible with Django 1.11 (and is also compatible with 2.2)
3.1.1 is the last compatible with Django 2.2
3.8.1 (latest) compatible with Django 3.2+

> django-extensions==1.7.8

2.2.9 is the last compatible with Django 1.11 (and is also compatible with 2.2)
3.1.5 is the last compatible with Django 2.2
3.2.1 (latest) compatible with Django 3.2+

> django-modeladmin-reorder==0.3.1

We're already on last release

> django-multiupload==0.6.1

We're already on last release

> django-oauth-toolkit==1.1.2

1.1.3 is the last compatible with Django 1.11 (but not clear if compatible with 2.2)
2.2.0 (latest) compatible with Django 2.2+

> django-object-actions==3.0.0
3.0.0 is the last compatible with Django 1.11 (and is also compatible with 2.2)
4.1.0 (latest) compatible with Django 3.2+ 

> django-ratelimit==2.0.0

3.0.1 is the last compatible with Django 1.11 (not 100% sure, should also work with Django 2.2)
4.0.0 (latest) compatible with Django 3.2+

> django-recaptcha==2.0.6

2.0.6 is the last compatible with Django 1.11 (and is also compatible with 2.2)
3.0.0 (latest) compatible with Django 2.2+

> django-redis==4.10.0

4.11.1 is the last compatible with Django 1.11 (and is also compatible with 2.2)
5.2.0 (latest) compatible with Django 2.2+

> django-silk==3.0.2

3.0.4 is the last compatible with Django 1.11 (and is also compatible with 2.2)
4.4.1 is the last compatible with Django 2.2
5.0.3 (latest) compatible with Django 3.2+

> djangorestframework-jsonp==1.0.2

We're already on last release

> djangorestframework-xml==2.0.0

We're already on last release

> djangorestframework-yaml==2.0.0

We're already on last release

> djangorestframework==3.9.4

3.10.0 is the last compatible with Django 1.11 (and is also compatible with 2.2)
3.13.1 is the last compatible with Django 2.2
3.14.0 (latest) compatible with Django 3.0+
